src/app/pages/my-virtrolio/my-virtrolio.component.ts

```diff
@@ -74,6 +93,42 @@ export class MyVirtrolioComponent implements OnInit {
     }
   }
 
+  shareLink() {
+    if (this.canShare()) {
+      this.navigator.share({
+        title: this.displayName + '\'s virtrolio!',
+        text: 'Visit this link to sign their virtrolio and send them a custom message!',
+        url: this.link,
+      })
+        .then(() => console.log('Successful share'))
+        .catch((error) => console.log('Error sharing', error));
+    } else {
+      alert('Share not supported on this device. Use a mobile device!');
+    }
+  }
+
+  /**
+   * Share the user's sharing link on the appropriate platform
+   * @param platform - platform on which to post the sharing link
+   */
+  postOnSocial(platform: string) {
+    const urlFriendlyLink = 'https%3A//virtrolio.web.app/signing?uid=' + this.authService.uid() + '%26key=' +
+      this.link.match(/key=([^&]*)/)[1];
+
+    const bodyText = 'Send%20me%20a%20message!%20';
```

Comment:
Twitter automatically adds a space, so this results in a double space. See my suggestion on line 126 below.
```suggestion
    const bodyText = 'Sign%20my%20virtual%20yearbook!';
```